src/platform-apps/channels/channel-view.tsx

```diff
@@ -1,27 +1,74 @@
+import classNames from 'classnames';
 import React from 'react';
 import { Message as MessageModel } from '../../store/messages';
 import { Message } from './message';
 
 export interface Properties {
   name: string;
   messages: MessageModel[];
+  onFirstMessageInViewport: (createdAt: MessageModel['createdAt']) => void;
 }
 
-export class ChannelView extends React.Component<Properties> {
+interface State {
+  hasScrolledToBottom: boolean;
+}
+
+export class ChannelView extends React.Component<Properties, State> {
+  channelViewRef: React.RefObject<HTMLDivElement>;
+  lastElementOnChannelViewRe: React.RefObject<HTMLDivElement>;
+  state = { hasScrolledToBottom: false };
+
+  constructor(props) {
+    super(props);
+
+    this.lastElementOnChannelViewRe = React.createRef();
+    this.channelViewRef = React.createRef();
+  }
+
+  componentDidMount() {
+    this.scrollToBottom();
+  }
+
+  setChannelViewRef = (ref) => {
+    this.channelViewRef = ref;
+  };
+
+  scrollToBottom = () => {
+    this.lastElementOnChannelViewRe.current?.scrollIntoView({ behavior: 'auto' });
+    this.setState({ hasScrolledToBottom: true });
+  };
```

Comment:
instead of managing this directly, i think i would add a ScrollContainer component that works based on content size rather than specific elements. if we only had to scroll once, then this makes sense, but the problem with handling reverse scrolling is that you need to update the scroll position any time the content changes at all. that means it needs to be updated not only when new messages are loaded, but also any time a message itself changes size. eg. when an image is loaded, when the content is replaced (mentions, emojis, etc.), or for any other reason.

you can look at the one in zero for ideas, but it is more aggressive than we will need here, it is made to support changes that occur outside of virtual dom updates - which we don't want to do, and is also super messy so i wouldn't copy the entire thing directly.